### PR TITLE
Add ldap_kerberos role

### LINKED ISSUE #5: Add role for ldap-kerberos
This role should deploy and configure openldap-server and MIT kerberos.

### PR DESCRIPTION
**Pull request by rpignolet**
#### Which issue(s) this PR fixes:

Fixes #5 

#### Additional comments:

To make clear that you license your contribution under
the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0) and that you give permission to [TOSIT](https://www.tosit.io/),
you have to acknowledge this by using the following check-box.

 - [x] I hereby declare this contribution to be licenced under the [Apache License Version 2.0, January 2004](http://www.apache.org/licenses/LICENSE-2.0)

 - [x] I hereby agree to grant [TOSIT](https://www.tosit.io/) a copyright license to use my contributions.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -52,3 +62,17 @@ ansible-playbook playbooks/certificates.yml
...
+
+On each host installs Kerberos clients and enable SSSD LDAP authentification.
+
+Create `ldap_groups` and `ldap_users` specified. For each users, a Kerberos principal is created and a keytab is generated in `/home/<username>/<username>.keytab` on  the`[users_keytab]` group hosts.
+
+```
+ansible-playbook playbooks/ldap_kerberos.yml
+```
+
+_After this, you can log in as the Kerberos admin from any VM with the command `kinit admin/admin` and the password `admin`. When using a user unix account on `[users_keytab]` group hosts, you can log in with `kinit -ki` which will use the keytab inside the home directory._
```
Comment (gboutry) [inline README.md:78]:
```suggestion
_After this, you can log in as the Kerberos admin from any VM with the command `kinit admin/admin` and the password `admin123`. When using a user unix account on `[users_keytab]` group hosts, you can log in with `kinit -ki` which will use the keytab inside the home directory._
```

The password seems to be `admin123`

File: roles/ldap_kerberos/templates/kdc.conf.j2 [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+[libdefaults]
+ dns_lookup_realm = false
+ ticket_lifetime = 24h
+ renew_lifetime = 7d
+ forwardable = true
+ rdns = false
+ pkinit_anchors = FILE:/etc/pki/tls/certs/ca-bundle.crt
+ default_realm = {{ realm }}
+ default_ccache_name = /tmp/krb5cc_%{uid}
+ canonicalize = true
```
Comment (gboutry) [inline roles/ldap_kerberos/templates/kdc.conf.j2:17]: Some of the variables are duplicated between `kdc.conf` and `krb5.conf`.
The kdc will merge `krb5.conf` and `kdc.conf`. Therefore there should be only the different / needed variables in `kdc.conf`.

https://web.mit.edu/kerberos/krb5-1.12/doc/admin/conf_files/kdc_conf.html

Comment (rpignolet) [reply roles/ldap_kerberos/templates/kdc.conf.j2:17]: Ok so according to the doc, we should put common configuration between client and server in `krb5.conf` and put in `kdc.conf` only stuff related to KDC server right ?

Comment (gboutry) [reply roles/ldap_kerberos/templates/kdc.conf.j2:17]: That's right

File: roles/ldap_kerberos/tasks/create_users_groups.yml [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+
+- name: Create LDAP users
+  shell: |
+    ldapadd -x -D {{ ldap_rootdn }} -w {{ ldap_rootpw }} << EOF
+    {{ ldap_user_ldif }}
+    EOF
+  register: reg_ldapadd
+  failed_when: |
+    reg_ldapadd is failed and
+    reg_ldapadd.rc != 68
```
[the remark this answers is not in the document]
Comment (rpignolet) [reply roles/ldap_kerberos/tasks/create_users_groups.yml]: done

File: roles/ldap_kerberos/templates/kdc.conf.j2 (hunk 2 above)
Comment (rpignolet) [reply roles/ldap_kerberos/templates/kdc.conf.j2]: done

File: README.md (hunk 1 above)
Comment (rpignolet) [reply README.md]: done

**Review by gboutry — APPROVED** [review]

**Review by leopaul36 — APPROVED** [review]

**Pull request merged**